FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW/main.c

```diff
@@ -0,0 +1,334 @@
+static volatile uint32_t ulCountOfItemsReceivedOnQueue = 0;
+static volatile uint32_t ulCountOfReceivedSemaphores = 0;
+
+int main( void )
+{
+    vTaskSuspendAll();
+
+	TimerHandle_t xExampleSoftwareTimer = NULL;
+	xQueue = xQueueCreate(     /* The number of items the queue can hold. */
+					mainQUEUE_LENGTH,
+					/* The size of each item the queue holds. */
+					sizeof( uint32_t ) );
+
+	/* Create the semaphore used by the FreeRTOS tick hook function and the
+	event semaphore task.  NOTE: A semaphore is used for example purposes,
+	using a direct to task notification will be faster! */
+	xEventSemaphore = xSemaphoreCreateBinary();
+
+	/* Create the queue receive task as described in the comments at the top
+	of this file. */
+	xTaskCreate(     /* The function that implements the task. */
+					prvQueueReceiveTask,
+					/* Text name for the task, just to help debugging. */
+					"Rx",
+					/* The size (in words) of the stack that should be created
+					for the task. */
+					configMINIMAL_STACK_SIZE,
+					/* A parameter that can be passed into the task.  Not used
+					in this simple demo. */
+					NULL,
+					/* The priority to assign to the task.  tskIDLE_PRIORITY
+					(which is 0) is the lowest priority.  configMAX_PRIORITIES - 1
+					is the highest priority. */
+					mainQUEUE_RECEIVE_TASK_PRIORITY,
+					/* Used to obtain a handle to the created task.  Not used in
+					this simple demo, so set to NULL. */
+					NULL );
+
+
+	/* Create the queue send task in exactly the same way.  Again, this is
+	described in the comments at the top of the file. */
+	xTaskCreate(     prvQueueSendTask,
+					"TX",
+					configMINIMAL_STACK_SIZE,
+					NULL,
+					mainQUEUE_SEND_TASK_PRIORITY,
+					NULL );
+
+
+	/* Create the task that is synchronised with an interrupt using the
+	xEventSemaphore semaphore. */
+	xTaskCreate(     prvEventSemaphoreTask,
+					"Sem",
+					configMINIMAL_STACK_SIZE,
+					NULL,
+					mainEVENT_SEMAPHORE_TASK_PRIORITY,
+					NULL );
+
+
+	/* Create the software timer as described in the comments at the top of
+	this file. */
+	xExampleSoftwareTimer = xTimerCreate(     /* A text name, purely to help
+											debugging. */
+											( const char * ) "LEDTimer",
+											/* The timer period, in this case
+											1000ms (1s). */
+											mainSOFTWARE_TIMER_PERIOD_MS,
+											/* This is a periodic timer, so
+											xAutoReload is set to pdTRUE. */
+											pdTRUE,
+											/* The ID is not used, so can be set
+											to anything. */
+											( void * ) 0,
+											/* The callback function that switches
+											the LED off. */
+											vExampleTimerCallback
+										);
+
+	/* Start the created timer.  A block time of zero is used as the timer
+	command queue cannot possibly be full here (this is the first timer to
+	be created, and it is not yet running). */
+	xTimerStart( xExampleSoftwareTimer, 0 );
+    xTaskResumeAll();
```

Comment:
As I can see you are mixing tabs with spaces. Please avoid it.

NOTE: this happens in multiple places in multiple files in this PR.